### PR TITLE
Remove generated files and update docs to generate files as they were

### PR DESCRIPTION
**Pull request by cpmeadors**
**- Description of the problem which is fixed/What is the use case**
There are files being tracked that are completely generated by make targets.  Fewer files makes maintaining the code easier. 

**- What I did**
Based on the operator-sdk docs [1], I have deleted the generated files from the repo.  One file required an additional command line change to generate the file exactly as it was before deleting.  That change was added to the DEVELOPMENT.md file.  I also add the relevant entries in the .gitignore to keep the files from being added back.

[1] https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/

This fixes: #209 and #208 

**- How to verify it**

If you execute the make file targets all files that were deleted should be recreated exactly the same:

make docker-build
make bundle
make bundle-build CHANNELS=candidate

**- Description for the changelog**

Generated files have been deleted from the repo

### CODE REVIEW

File: docs/DEVELOPMENT.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ make docker-push
 
 ## Building Operator bundle image
 ```
-make bundle
+make bundle CHANNELS=candidate
```
Comment (jensfr) [inline docs/DEVELOPMENT.md:51]: candidate is also what we use in our example subscription in config/samples/deploy, so I think this is a good change

**Review by jensfr — APPROVED** [review]
Looks good to me! Thank you Cameron!

**Review by pmores — APPROVED** [review]
Thanks @cpmeadors !

**Comment by pmores** [comment]
Merging is blocked though because CI is failing which seems to be due to missing `DeepCopyObject()` methods in `KataConfig`, `KataConfigList` and possibly others.  I suspect that our current build sequence is missing something to regenerate `zz_generated.deepcopy.go` which is removed by this PR.

**Comment by pmores** [comment]
What's missing seems to be a `make generate` invocation. Unfortunately I've no idea ATM how CI works but from looking at the `Makefile`, the `generate` target is also invoked by `test`, `build` and `run` targets as a prerequisite. So it looks like none of these targets are invoked by CI.

**Pull request merged**